go-controller/pkg/clustermanager/default_network_cluster_controller.go

```diff
@@ -0,0 +1,416 @@
+package clustermanager
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+
+	hotypes "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	houtil "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/clustermanager/subnetallocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	objretry "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	kapi "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+type networkClusterControllerBase struct {
+	kube         kube.Interface
+	watchFactory *factory.WatchFactory
+	stopChan     chan struct{}
+	wg           *sync.WaitGroup
+
+	// node events factory handler
+	nodeHandler *factory.Handler
+
+	// retry framework for nodes
+	retryNodes *objretry.RetryFramework
+}
+
+// defaultNetworkClusterController is the cluster controller for the default network
+// for all the nodes. Listens to the node events and allocates subnet from the
+// default cluster subnet pool.
+type defaultNetworkClusterController struct {
+	networkClusterControllerBase
+	defaultNetworkController *NodeNetworkController
+
+	hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+}
+
+func newDefaultNetworkClusterController(ovnClient *util.OVNClientset, wf *factory.WatchFactory) *defaultNetworkClusterController {
+
+	kube := &kube.Kube{
+		KClient: ovnClient.KubeClient,
+	}
+
+	stopChan := make(chan struct{})
+	wg := &sync.WaitGroup{}
+
+	var hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+	if config.HybridOverlay.Enabled {
+		hybridOverlaySubnetAllocator = subnetallocator.NewHostSubnetAllocator()
+	}
+	ncm := &defaultNetworkClusterController{
+		networkClusterControllerBase: networkClusterControllerBase{
+			kube:         kube,
+			watchFactory: wf,
+			stopChan:     stopChan,
+			wg:           wg,
+		},
+		defaultNetworkController: newNodeNetworkController(kube, wf, types.DefaultNetworkName),
+
+		hybridOverlaySubnetAllocator: hybridOverlaySubnetAllocator,
+	}
+
+	ncm.initRetryFramework()
+	return ncm
+}
+
+func (ncm *defaultNetworkClusterController) initRetryFramework() {
+	ncm.retryNodes = ncm.newRetryFramework(factory.NodeType)
+}
+
+// Start initializes the default network subnet allocator ranges
+// and hybrid network subnet allocator ranges if hybrid overlay is enabled.
+func (ncm *defaultNetworkClusterController) Start() error {
```

Comment:
I'll rename the function to "Init" instead of InitSubnetAllocator, so that we have Init+Run.

Let me know if you've a strong opinion on using InitSubnetAllocator.

Reply:
Nope. Looks good.